File: Stage.java
package omo;

class Stage
{
    private String[] text = new String[UI.MAX_LINES];
    private Stage nextStage;

    Stage(Stage nextStage, String[] text)
    {
        this.nextStage = nextStage;
        this.text = text;
    }

    Stage(String[] text)
    {
        this.text = text;
    }

    Stage(Stage nextStage)
    {
        this.nextStage = nextStage;
    }

    void operate()
    {

    }

    void operate(UI ui)
    {

    }

    void run(UI ui)
    {
        ui.setText(getText());
        operate();
    }

    /**
     * @return the stage text
     */
    public String[] getText()
    {
        return text;
    }

    /**
     * @param text the stage text to set
     */
    public void setText(String... text)
    {
        this.text = text;
    }
}

File: omo/Stage.kt
package omo

class Stage(
        val text: Text = Text.empty,
        val operations: () -> Unit = {},
        val nextStage: StageID? = StageID.END_GAME,
        val actionGroupName: String = "",
        val actions: List<Action> = listOf<Action>(),
        val duration: Int = 0,
        val score: Int = 0,
        val cheatLists: List<List<Cheat>> = listOf(Cheat.global),
        val nextStagePriority: NextStagePriority = NextStagePriority.ACTION
) {
    class Text(vararg lines: Line) {
        class Line(
                val line: String,
                val italic: Boolean = false
        ) {
            companion object {
                val empty = Line("")
            }
        }

        val lines = lines

        companion object {
            val empty = Text(Line(""))
        }
    }

    enum class NextStagePriority {
        ACTION,
        DEFINED
    }
}